src/kedro_workbench/pipelines/docstore_feature_engineering_pm/nodes.py

```diff
@@ -327,58 +352,70 @@ def batch_update_new_features_patch(data):
-                {'$set': update_data},
-                upsert=True
+            try:
+                update_result = mongo.collection.update_one(
+                    {"metadata.id": record_id}, {"$set": update_data}, upsert=True
                 )
-            print_mongo_result_properties(update_result, verbose=False)
-            num_matched += update_result.matched_count
-            num_affected += update_result.modified_count
+                print_mongo_result_properties(update_result, verbose=False)
+                num_matched += update_result.matched_count
+                num_affected += update_result.modified_count
+            except Exception as e:
+                logger.error(f"Error updating MongoDB record {record_id}: {e}")
+                raise
         logger.info(
             f"Completed patch management feature engineering save to Mongo."
             f"{num_matched} records matched and {num_affected} records affected."
-            )
+        )
         return True
 
 
 def remove_mongo_duplicates_patch(update_flag):
     if update_flag:
-        cursor = mongo.collection.aggregate([
-            {
-                "$match": {"metadata.collection": "patch_management"}
-            },
-            {
-                "$group": {
-                    "_id": {"collection": "$metadata.collection", "id": "$metadata.id"},
-                    "count": {"$sum": 1},
-                    "duplicates": {"$push": "$_id"}
-                }
-            },
-            {
-                "$match": {"count": {"$gt": 1}}
-            }
-            ])
-        dups_list = list(cursor)
-        if len(dups_list) > 0:
-            print(f"found {len(dups_list)} duplicates in docstore patch_management...")
-            deletion_summary = remove_duplicates(mongo.collection, dups_list)
-            # print(f"Deletion Summary: {deletion_summary}")
+        try:
+            cursor = mongo.collection.aggregate(
+                [
+                    {"$match": {"metadata.collection": "patch_management"}},
+                    {
+                        "$group": {
+                            "_id": {
+                                "collection": "$metadata.collection",
+                                "id": "$metadata.id",
+                            },
+                            "count": {"$sum": 1},
+                            "duplicates": {"$push": "$_id"},
+                        }
+                    },
+                    {"$match": {"count": {"$gt": 1}}},
+                ]
+            )
+            dups_list = list(cursor)
+            if len(dups_list) > 0:
+                print(
+                    f"found {len(dups_list)} duplicates in docstore patch_management..."
+                )
+                deletion_summary = remove_duplicates(mongo.collection, dups_list)
+        except Exception as e:
+            logger.error(f"Error removing MongoDB duplicates: {e}")
+            raise
```

Comment:
can you add a logger statement here instead of removing the assignment?
```suggestion
            if len(dups_list) > 0:
                print(
                    f"found {len(dups_list)} duplicates in docstore patch_management..."
                )
                deletion_summary = remove_duplicates(mongo.collection, dups_list)
                logger.error(f"Removed duplicates summary:\n {deletion_summary}")

```